api/src/services/report/smsparser.js

```diff
@@ -237,6 +260,28 @@ exports.parse = (def, doc) => {
     }
   }
 
+  if(hasBSDateField) {
+    let bsYear;
+    let bsMonth = 1;
+    let bsDay = 1;
+    for (const k of Object.keys(def.fields)) {
+      switch (def.fields[k].type) {
+      case 'bsYear':
+        bsYear = msgData[k];
+        break;
+      case 'bsMonth':
+        bsMonth = msgData[k];
+        break;
+      case 'bsDay':
+        bsDay = msgData[k];
+        break;
+      }
+    }
+    
+    formData.lmp_date = bsToEpoch(bsYear, bsMonth, bsDay);
```

Comment:
Thanks for pointing out this. I agree that we should avoid hardcoded field names as much as possible.

> use a different field type to aggregate the broken down BS date

We also have a new `bsDate` field type. Shall we use it to aggregate the date parts?
Shall we also check for the field type in `transitions/registration/getLMPDate` instead of the hard-coded names?
Please let me know whether you had a different idea.

Reply:
> We also have a new bsDate field type. Shall we use it to aggregate the date parts?

My gut feeling is that we'd have to use a different type to keep things easy to understand, and never have questions like: where is this "bsDate" value coming from? If you have a form with both `bsDate` and `bsYear`, which one would "win"? 

Parsing is going to be awkward, because you'll need to put the "splitBsDate" field last in the form definition every time. :(

Reply:
Thanks! I've added a new field type `bsAggreDate` for this purpose.
I've added an e2e test in `tests/e2e/translations/nepali-dates-and-numbers.wdio-spec.js` using this type.